util/typeutil/stringInt.go

```diff
@@ -0,0 +1,32 @@
+package typeutil
+
+import (
+	"encoding/json"
+	"strconv"
+)
+
+type StringInt int
+
+func (st *StringInt) UnmarshalJSON(b []byte) error {
+	//convert the bytes into an interface as this will help us check the type of our value
+	var item interface{}
+	if err := json.Unmarshal(b, &item); err != nil {
+		return err
+	}
```

Comment:
If you're up for running a benchmark test, I am curious if there is a performance cost for unmarshalling to an empty interface (one unmarshal call total), instead of attempting to unmarshal to specific types (one unmarshal call for the target type). Perhaps its even faster to use jsonparser Get methods for int and string?

Reply:
Using JsonParser seems to be the most efficient to use while doing json.Unmarshal. The only catch is when string is provided as input we have to remove double quotes in Unmarshal function. Here is the how I implemented the benchmarking code. Here are the benchmarking results on my local machine - 

```
BenchmarkUnmarshalJSONUsingEmptyInterfaceWithInputAsString-16                	 1761681	       677.5 ns/op
BenchmarkUnmarshalJSONUsingEmptyInterfaceWithInputAsString-16                	 1658487	       710.2 ns/op
BenchmarkUnmarshalJSONUsingEmptyInterfaceWithInputAsString-16                	 1775126	       685.4 ns/op
BenchmarkUnmarshalJSONUsingEmptyInterfaceWithInputAsString-16                	 1744000	       675.1 ns/op
BenchmarkUnmarshalJSONUsingEmptyInterfaceWithInputAsString-16                	 1771329	       680.5 ns/op
BenchmarkUnmarshalJSONUsingEmptyInterfaceWithInputAsInt-16                   	 1715920	       697.6 ns/op
BenchmarkUnmarshalJSONUsingEmptyInterfaceWithInputAsInt-16                   	 1744148	       724.0 ns/op
BenchmarkUnmarshalJSONUsingEmptyInterfaceWithInputAsInt-16                   	 1720200	       704.9 ns/op
BenchmarkUnmarshalJSONUsingEmptyInterfaceWithInputAsInt-16                   	 1626032	       699.2 ns/op
BenchmarkUnmarshalJSONUsingEmptyInterfaceWithInputAsInt-16                   	 1714712	       694.2 ns/op
BenchmarkUnmarshalJSONUsingJsonParserWithInputAsString-16                    	 2586522	       447.4 ns/op
BenchmarkUnmarshalJSONUsingJsonParserWithInputAsString-16                    	 2704970	       450.6 ns/op
BenchmarkUnmarshalJSONUsingJsonParserWithInputAsString-16                    	 2475841	       451.7 ns/op
BenchmarkUnmarshalJSONUsingJsonParserWithInputAsString-16                    	 2574088	       463.3 ns/op
BenchmarkUnmarshalJSONUsingJsonParserWithInputAsString-16                    	 2684553	       452.3 ns/op
BenchmarkUnmarshalJSONUsingJsonParserWithInputAsInt-16                       	 2616318	       447.0 ns/op
BenchmarkUnmarshalJSONUsingJsonParserWithInputAsInt-16                       	 2645331	       453.4 ns/op
BenchmarkUnmarshalJSONUsingJsonParserWithInputAsInt-16                       	 2644465	       451.9 ns/op
BenchmarkUnmarshalJSONUsingJsonParserWithInputAsInt-16                       	 2637205	       461.6 ns/op
BenchmarkUnmarshalJSONUsingJsonParserWithInputAsInt-16                       	 2617648	       450.8 ns/op
BenchmarkUnmarshalJSONUsingJsonParserAndStringReplaceWithInputAsString-16    	 2309619	       511.4 ns/op
BenchmarkUnmarshalJSONUsingJsonParserAndStringReplaceWithInputAsString-16    	 2301285	       512.1 ns/op
BenchmarkUnmarshalJSONUsingJsonParserAndStringReplaceWithInputAsString-16    	 2358621	       509.2 ns/op
BenchmarkUnmarshalJSONUsingJsonParserAndStringReplaceWithInputAsString-16    	 2316736	       518.6 ns/op
BenchmarkUnmarshalJSONUsingJsonParserAndStringReplaceWithInputAsString-16    	 2313619	       534.0 ns/op
BenchmarkUnmarshalJSONUsingJsonParserAndStringReplaceWithInputAsInt-16       	 2541432	       466.0 ns/op
BenchmarkUnmarshalJSONUsingJsonParserAndStringReplaceWithInputAsInt-16       	 2620980	       466.5 ns/op
BenchmarkUnmarshalJSONUsingJsonParserAndStringReplaceWithInputAsInt-16       	 2603096	       464.6 ns/op
BenchmarkUnmarshalJSONUsingJsonParserAndStringReplaceWithInputAsInt-16       	 2617431	       464.7 ns/op
BenchmarkUnmarshalJSONUsingJsonParserAndStringReplaceWithInputAsInt-16       	 2596263	       461.2 ns/op
BenchmarkUnmarshalJSONUsingDataTypeWithInputAsString-16                      	 2394945	       518.6 ns/op
BenchmarkUnmarshalJSONUsingDataTypeWithInputAsString-16                      	 2393102	       507.8 ns/op
BenchmarkUnmarshalJSONUsingDataTypeWithInputAsString-16                      	 2366752	       500.1 ns/op
BenchmarkUnmarshalJSONUsingDataTypeWithInputAsString-16                      	 2366858	       499.0 ns/op
BenchmarkUnmarshalJSONUsingDataTypeWithInputAsString-16                      	 2379032	       559.0 ns/op
BenchmarkUnmarshalJSONUsingDataTypeWithInputAsInt-16                         	 2462937	       487.6 ns/op
BenchmarkUnmarshalJSONUsingDataTypeWithInputAsInt-16                         	 2549624	       485.0 ns/op
BenchmarkUnmarshalJSONUsingDataTypeWithInputAsInt-16                         	 2558164	       458.3 ns/op
BenchmarkUnmarshalJSONUsingDataTypeWithInputAsInt-16                         	 2648469	       447.8 ns/op
BenchmarkUnmarshalJSONUsingDataTypeWithInputAsInt-16                         	 2622322	       464.6 ns/op
```

Reply:
Excellent! Thank you for running the benchmarks so we can decide based on actual performance data. I summarized the different approaches:

```
BenchmarkUnmarshalJSONUsingEmptyInterface              694.86 ns/op average
BenchmarkUnmarshalJSONUsingJsonParser                  453.00 ns/op average
BenchmarkUnmarshalJSONUsingJsonParserAndStringReplace  490.83 ns/op average
BenchmarkUnmarshalJSONUsingDataType                    492.78 ns/op average
```

The `EmptyInterface` clearly performs the worst, so I think we should rule that out. I'm not surprised by this finding since reflection is still expensive in Go. 

`JsonParser` is the fastest and might be a good choice. You may even be able to further improve performance by replacing `bytes.HasPrefix` with a hardcoded check of the first character (if there is a first character). It seems `JsonParserAndStringReplace` is the lower performance variant and should be excluded as an option.

The `DataType` approach has good performance if we want to keep with the stock json package, but I see no compelling reason for doing so. That package has improved validation checks, but by the time we've reached this part of code the json has already been validated.

Reply:
thanks @SyntaxNode! Based on your suggestion above, I was able to achieve some more performance improvement by directly checking the first character. Here is the revised version. Please take a look - https://github.com/prebid/prebid-server/pull/2617/commits/9e4089964012932b327478fc68b37b411a51bdcf